### PR TITLE
Fix clientFilesQuestion for shared questions

### PR DESCRIPTION
**Pull request by SethPoulsen**

### CODE REVIEW

File: apps/prairielearn/src/pages/clientFilesQuestion/clientFilesQuestion.js [hunk 1]
```diff
@@ -1,28 +1,32 @@
...
-  const coursePath = chunks.getRuntimeDirectoryForCourse(res.locals.course);
-  const chunk = {
-    type: 'question',
-    questionId: res.locals.question.id,
-  };
-  chunks.ensureChunksForCourse(res.locals.course.id, chunk, (err) => {
-    if (ERR(err, next)) return;
+      });
+    }
+    const course = await getQuestionCourse(res.locals.question, res.locals.course).then();
```
Comment (nwalters512) [inline apps/prairielearn/src/pages/clientFilesQuestion/clientFilesQuestion.js:21]: Why the `.then()`?

**Review by nwalters512 — COMMENTED** [review]
One question before I approve!

File: apps/prairielearn/src/pages/clientFilesQuestion/clientFilesQuestion.js (hunk 1 above)
Comment (SethPoulsen) [reply apps/prairielearn/src/pages/clientFilesQuestion/clientFilesQuestion.js]: Oh that's very funny. I was momentarily thinking about using promises before I just decided to refactor the handler to be async, this is a remnant of that. 

I have removed it.

File: apps/prairielearn/src/pages/clientFilesQuestion/clientFilesQuestion.js [hunk 2]
```diff
@@ -1,28 +1,32 @@
...
-  const coursePath = chunks.getRuntimeDirectoryForCourse(res.locals.course);
-  const chunk = {
-    type: 'question',
-    questionId: res.locals.question.id,
-  };
-  chunks.ensureChunksForCourse(res.locals.course.id, chunk, (err) => {
-    if (ERR(err, next)) return;
+      });
+    }
+    const course = await getQuestionCourse(res.locals.question, res.locals.course);
```
Comment (nwalters512) [inline apps/prairielearn/src/pages/clientFilesQuestion/clientFilesQuestion.js:21]: Refresh my memory: by the time we get to this point, we've correctly authenticated the the current user/course has access to this question via sharing, correct? I think that's true, but I always like to triple-check with auth code.

File: apps/prairielearn/src/pages/clientFilesQuestion/clientFilesQuestion.js [hunk 3]
```diff
@@ -1,28 +1,32 @@
...
+    const coursePath = chunks.getRuntimeDirectoryForCourse(course);
+
+    /** @type {chunks.QuestionChunk} */
+    const chunk = {
+      type: 'question',
+      questionId: res.locals.question.id,
+    };
+    await chunks.ensureChunksForCourseAsync(res.locals.course.id, chunk);
```
Comment (nwalters512) [inline apps/prairielearn/src/pages/clientFilesQuestion/clientFilesQuestion.js:29]: I believe we might actually want this?

```suggestion
    await chunks.ensureChunksForCourseAsync(course.id, chunk);
```

Comment (SethPoulsen) [reply apps/prairielearn/src/pages/clientFilesQuestion/clientFilesQuestion.js:29]: ya, I think that's right. I've fixed it.

File: apps/prairielearn/src/pages/clientFilesQuestion/clientFilesQuestion.js (hunk 2 above)
Comment (SethPoulsen) [reply apps/prairielearn/src/pages/clientFilesQuestion/clientFilesQuestion.js:21]: That's right, auth is all taken care of. 

Whenever one of these URLs is accessed through an `instance_question`, which is any time a student accesses it, the instance question was created from an assessment question, which was created at sync time after validating that the course has permissions to use this question.

If an instructor accesses this through a link that is not associated with a question instance (say, in a question preview), then we validate in the middleware that the question has been shared with the course.

**Review by nwalters512 — APPROVED** [review]

**Pull request merged**